yarn-project/circuits.js/src/structs/rollup/new_l1_to_l2_messages.ts

```diff
@@ -0,0 +1,79 @@
+import { BufferReader } from '@aztec/foundation/serialize';
+import times from 'lodash.times';
+import { assertMemberLength } from '../../index.js';
+import { serializeToBuffer } from '../../utils/serialize.js';
+import { NUMBER_OF_L1_L2_MESSAGES_PER_ROLLUP } from '../constants.js';
+import { Fr } from '../index.js';
+
+/**
+ * Pads an array to the desired length with zero values.
+ * @param arr - The array to pad.
+ * @param length - The length to pad to.
+ * @returns
+ */
+function padToLength(arr: Fr[], length: number): Fr[] {
+  if (arr.length > length) {
+    throw Error('Array is already longer than the desired length');
+  }
+  if (arr.length === length) {
+    return arr;
+  }
+  return arr.concat(Array(length - arr.length).fill(Fr.zero()));
+}
+
+/**
+ * Type wrapper for L1 to L2 messages.
+ */
+export class NewL1ToL2Messages {
+  /**
+   * Commitments to the new l1 to l2 messages.
+   */
+  public l1ToL2Messages: Fr[];
```

Comment:
-1 to the name, we'll end up with `struct.newL1ToL2Messages.l1ToL2Messages` whenever we want to access them. Maybe just a generic `items` or `values`..?